ares.h

```diff
@@ -368,8 +368,8 @@ struct iovec;
 struct ares_socket_functions {
    ares_socket_t(*asocket)(int, int, int, void *);
    int(*aclose)(ares_socket_t, void *);
-   int(*aconnect)(ares_socket_t, const struct sockaddr *, socklen_t, void *);
-   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, socklen_t *, void *);
+   int(*aconnect)(ares_socket_t, const struct sockaddr *, ares_socklen_t, void *);
+   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, ares_socklen_t *, void *);
    ssize_t(*asendv)(ares_socket_t, const struct iovec *, int, void *);
```

Comment:
@ChristianAmmer yeah, you're probably right about that as well, I didn't hit that build error on any of my systems, but it probably would fail if I built the test cases on all systems.  That issue is a little less easy to fix since there is no   ares_ssize_t  datatype.